Doc/howto/sorting.rst

```diff
@@ -272,6 +275,71 @@ to make it usable as a key function::
 
     sorted(words, key=cmp_to_key(strcoll))  # locale-aware sort order
 
+Strategies For Unorderable Types and Values
+===========================================
+
+A number of type and value issues can arise when sorting.
+Here are some strategies that can help:
+
+* Convert non-comparable input types to strings prior to sorting:
+
+.. doctest::
+
+   >>> data = ['twelve', '11', 10]
+   >>> sorted(map(str, data))
+   ['10', '11', 'twelve']
+
+This is needed because most cross-type comparisons raise a
+:exc:`TypeError`.
+
+* Remove special values prior to sorting:
+
+.. doctest::
+
+   >>> from math import isnan
+   >>> from itertools import filterfalse
+   >>> data = [3.3, float('nan'), 1.1, 2.2]
+   >>> sorted(filterfalse(isnan, data))
+   [1.1, 2.2, 3.3]
+
+This is needed because the `IEEE-754 standard
+<https://en.wikipedia.org/wiki/IEEE_754>`_ specifies that, "Every NaN
+shall compare unordered with everything, including itself."
+
+Likewise, ``None`` can be stripped from datasets as well:
+
+.. doctest::
+
+   >>> data = [3.3, None, 1.1, 2.2]
+   >>> sorted(x for x in data if x is not None)
+   [1.1, 2.2, 3.3]
+
+This is needed because ``None`` is not comparable to other types.
+
+* Convert mapping types into sorted item lists before sorting:
+
+.. doctest::
+
+   >>> from operator import methodcaller
+   >>> data = [{'a': 1}, {'b': 2}]
+   >>> list(map(dict, sorted(map(methodcaller('items'), data))))
```

Comment:
Why not use `key=` here?
```suggestion
   >>> sorted(data, key=methodcaller('items'))
```

Reply:
I don't think this works.  IIRC, an items-view implements set-like comparisons which are a subset/superset tests instead of a total ordering.  Also, the intent here was to sort the key/value pairs to so that dicts that compare equal regardless of key ordering will appear side-by-side in the output.

Reply:
It does work here, but it does rely on the set operations and will not work for all dicts. However, that is also true for your version:

```
>>> data = [{'a': 1}, {'b': 2}, {'a': 3}]
>>> list(map(dict, sorted(map(methodcaller('items'), data))))
[{'a': 1}, {'b': 2}, {'a': 3}]
>>> sorted(data, key=methodcaller('items'))
[{'a': 1}, {'b': 2}, {'a': 3}]
```

To work correctly, we need to sort the itemsviews:

```
>>> sorted(data, key=lambda d: sorted(d.items()))
[{'a': 1}, {'a': 3}, {'b': 2}]
```

Reply:
That makes sense.